More vehicle variety in convoy mission

wersal454 opened the pull request:
## What type of PR is this?
1. [ ] Bug
2. [x] Change
3. [x] Enhancement
4. [x] Miscellaneous

### What have you changed and why?
Information:

replaced AA vehicle with randomization with vehiclesArmor (i.e. it can be tank, apc, light tank, etc.)

now troops trucks can be militia or regular military depending on the war level

also lead vehicle can be militia or military depending on the war level

added repair type of convoy 

### Please specify which Issue this PR Resolves (If Applicable).
"This PR closes #XXXX!"

### Please verify the following.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:

jaj22 committed:
Improve garage source checking

Bob-Murphy committed:
Update README.md

Bob-Murphy committed:
Merge pull request #2860 from jaj22/garage-improve-source-checking

Improve garage source checking

Bob-Murphy committed:
logistics_localization (#2881)

* logistics_localization

* Shortened names, added missing components

---------

Co-authored-by: Bob-Murphy <Bob-Murphy@users.noreply.github.com>
Co-authored-by: Whiskeysim(Hazey) <1194692+MeltedPixel@users.noreply.github.com>

jaj22 committed:
Enable rebel AI to use rifle+muzzle combo grenade launchers (#2899)

* Enable rebel AI to use rifle+muzzle combo grenade launchers

* Improve readability of special GL filtering

wersal454 committed:
Added APEX dlc UAV

wersal454 committed:
Added JETS dlc UAV

wersal454 committed:
Added JETS dlc UAV

Note*
Noticed that Arid template has "B_UAV_02_dynamicLoadout_F" and others have "B_UAV_02_CAS_F" please check

wersal454 committed:
Added JETS dlc UAV

Note*
Noticed that Nato Arid template have "B_UAV_02_dynamicLoadout_F" while others have "B_UAV_02_CAS_F" please check

wersal454 committed:
Added JETS dlc UAV

Note*
Noticed that Nato Arid template have "B_UAV_02_dynamicLoadout_F" while others have "B_UAV_02_CAS_F" please check

wersal454 committed:
Merge branch 'unstable' into stable

wersal454 committed:
Merge pull request #17 from wersal454/stable

Stable

wersal454 committed:
Merge pull request #18 from SilenceIsFatto/stable

Merge pull request #161 from SilenceIsFatto/unstable

wersal454 committed:
Merge pull request #21 from SilenceIsFatto/unstable

Unstable

wersal454 committed:
Merge pull request #23 from SilenceIsFatto/unstable

Unstable

wersal454 committed:
Delete Sonic.ogg

wersal454 committed:
Create Sonic.ogg

wersal454 committed:
bebe

wersal454 committed:
Merge branch 'QRF-orbital' into unstable

wersal454 committed:
convoy variety

wersal454 committed:
more convoy vehicle variety

untested

wersal454 committed:
Update fn_convoy.sqf

jwoodruff40 committed:
Select faction-defined supply trucks for money and supply convoys before falling back to rebel faction-defined trucks (A3 vanilla box van in *almost* all templates >.<)

jwoodruff40 committed:
Use appropriate faction obj vehicles for supply and money convoys

jwoodruff40 committed:
Fix termination conditions

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -262,7 +290,20 @@ for "_i" from 1 to _countX do
 // Lead vehicle
 sleep 2;
-private _typeVehX = selectRandom (if (_sideX == Occupants && random 4 < tierWar) then {_faction get "vehiclesPolice"} else {_faction get "vehiclesLightArmed"});
+private _typeVehX = selectRandom (if (_sideX == Occupants && random 4 < tierWar) then {
+        _faction get "vehiclesPolice"
+    } else {
+        if (tierWar < 5) then {
+            (_faction get "vehiclesMilitiaLightArmed")
+        } else {
+            if (tierWar < 7) then {
+                (_faction get "vehiclesLightArmed") + (_faction get "vehiclesMilitiaLightArmed")
+            } else {
+                (_faction get "vehiclesLightArmed")
+            }
+        };
+    }
+); ///maybe a proper war level check?
```
Maybe something like this?

```suggestion
private _typeVehX = selectRandom ( switch true do {
    case (_sideX == Occupants && random 4 < tierWar): { (_faction get "vehiclesPolice") };
    case (tierWar < 5): { (_faction get "vehiclesMilitiaLightArmed") };
    case (tierWar < 7): { (_faction get "vehiclesLightArmed") + (_faction get "vehiclesMilitiaLightArmed") };
    default { (_faction get "vehiclesLightArmed") };
});
```

It's doing the same thing, in half the lines and without as much nesting (of course, I'm also guilty of generally nesting a lot...)

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -86,43 +90,67 @@ switch (toLowerANSI _convoyType) do
         _taskTitle = localize "STR_A3A_Missions_AS_Convoy_task_header_prisoners";
         _taskIcon = "run";
-        _typeVehObj = selectRandom (_faction get "vehiclesTrucks");
+        _typeVehObj = selectRandom (if (tierWar < 5) then {(_faction get "vehiclesMilitiaTrucks")
+        } else {
+            if (tierWar < 7) then {
+                (_faction get "vehiclesMilitiaTrucks") + (_faction get "vehiclesTrucks")
+            } else {
+                (_faction get "vehiclesTrucks")
+            }
+        };)
```
Similar to comment below...

```suggestion
_typeVehObj = selectRandom ( switch true do {
            case (tierWar < 5): { (_faction get "vehiclesMilitiaTrucks") };
            case (tierWar < 7): { (_faction get "vehiclesMilitiaTrucks") + (_faction get "vehiclesTrucks") };
            default { (_faction get "vehiclesTrucks") };
        });
```

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -86,43 +90,67 @@ switch (toLowerANSI _convoyType) do
         _taskTitle = localize "STR_A3A_Missions_AS_Convoy_task_header_reinf";
         _taskIcon = "meet";
-        _typeVehObj = selectRandom (_faction get "vehiclesTrucks");
+        _typeVehObj = selectRandom (if (tierWar < 5) then {(_faction get "vehiclesMilitiaTrucks")
+        } else {
+            if (tierWar < 7) then {
+                (_faction get "vehiclesMilitiaTrucks") + (_faction get "vehiclesTrucks")
+            } else {
+                (_faction get "vehiclesTrucks")
+            }
+        };)
```
Similar to comment below...

```suggestion
_typeVehObj = selectRandom ( switch true do {
            case (tierWar < 5): { (_faction get "vehiclesMilitiaTrucks") };
            case (tierWar < 7): { (_faction get "vehiclesMilitiaTrucks") + (_faction get "vehiclesTrucks") };
            default { (_faction get "vehiclesTrucks") };
        });
```

jwoodruff40 commented:
Left a couple of suggestions - I'm working on a [similar PR](https://github.com/wersal454/A3-Antistasi/pull/30) that makes money and supplies convoys more faction appropriate, so wanted to get this one to stop throwing script errors

jwoodruff40 committed:
Merge branch 'improve-convoy-vehicle-selection' into pr/wersal454/428

jwoodruff40 committed:
Rework war level vehicle selection

wersal454 commented:
yeah, always forget that switch exists, that will do

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: John Woodruff <rocketpanda40@gmail.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: John Woodruff <rocketpanda40@gmail.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: John Woodruff <rocketpanda40@gmail.com>

wersal454 committed:
Merge branch 'patch-50' into pr/wersal454/428

Silence-112 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -86,43 +90,61 @@ switch (toLowerANSI _convoyType) do
+        _taskIcon = "box";
+        private _supplyVehicles = (FactionGet(reb, "vehiclesCivSupply"));
+        private _medicalVehicles = _faction get "vehiclesMedical";
+        _vehiclePool = append [_supplyVehicles,_medicalVehicles];
```
`append [array, array]`?

Append works by appending one array to another, e.g `array1 append array2`

What I assume you want here is
```suggestion
        _vehiclePool = (_supplyVehicles + _medicalVehicles);
```

Silence-112 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -368,6 +395,24 @@ if (_convoyType isEqualTo "Fuel") then
 
+if (_convoyType isEqualTo "Repair") then
+{
+    waitUntil {sleep 1; (time > _timeout) or (_vehObj distance _posDest < _arrivalDist) or (not alive _vehObj) or (side group driver _vehObj != _sideX)};
```
Have you tried lazy evaluation here? Would probably be better scheduling wise

Example
```suggestion
    waitUntil {sleep 1; (time > _timeout) || {(_vehObj distance _posDest < _arrivalDist) or (not alive _vehObj) or (side group driver _vehObj != _sideX)}};
```

Silence-112 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -8,6 +8,7 @@ params ["_mrkDest", "_mrkOrigin", ["_convoyType", ""], ["_resPool", "legacy"], [
 private _difficult = if (random 10 < tierWar) then {true} else {false};
 private _sideX = if (sidesX getVariable [_mrkOrigin,sideUnknown] == Occupants) then {Occupants} else {Invaders};
 private _faction = Faction(_sideX);
+private _sideshort =  if (sidesX getVariable [_mrkOrigin,sideUnknown] == Occupants) then {occ} else {inv};
```
occ/inv only exist in the FactionGet macros, no?

```sqf
#define FactionGet(FAC, VAR) (A3A_faction_##FAC get VAR)
```

Also isn't this redundant given _sideX and _faction are already defined, so you can just use `_faction get "XYZ"`?

```suggestion
```

Silence-112 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -86,43 +90,61 @@ switch (toLowerANSI _convoyType) do
         _taskTitle = localize "STR_A3A_Missions_AS_Convoy_task_header_armor";
         _taskIcon = "destroy";
-        _typeVehObj = selectRandom (_faction get "vehiclesAA");
+        _typeVehObj = selectRandom (FactionGet(_sideshort, "vehiclesArmor"));
```
```suggestion
        _typeVehObj = selectRandom (_faction get "vehiclesArmor");
```

Silence-112 reviewed (commented):
I assume that you plan on merging [#30](https://github.com/wersal454/A3-Antistasi/pull/30)?

jwoodruff40 committed:
Use existing _faction get XXX for armored vehicles

jwoodruff40 committed:
Cargo and vehicle selection fixes.

Get vehiclesCivSupply from the rebel template, not occ/inv.

Don't attempt to load cargo into vehicle if it came from
vehiclesCivSupply (they don't have cargo nodes because they're enclosed
vehicles by design).

Set the objective object that must be stolen / destroyed to either the
vehicle OR the cargo, depending on if cargo was loaded or not (see
above).

Ensure termination conditions work correctly.

Only delete the objective object in termination if it's a cargo object.
Otherwise, let A3A_fnc_VEHdespawner deal with the vehicle itself.

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Merge branch 'patch-50' into pr/wersal454/428

jwoodruff40 committed:
Use all faction-defined vehicles

instead of inheriting defaults.

Add variety of objective vehicle by war level: civilian --> militia -->
military ( + medical-specific)

wersal454 committed:
remove the boxer

jwoodruff40 committed:
Simplify convoy objective vehicle selection

Civilian vehicles only - vehiclesCivIndustrial / vehiclesCivSupply for
money convoys, vehiclesCivMedical / vehiclesCivSupply for supply convoys

jwoodruff40 committed:
Remove vehiclesCivSupply from defaults entirely

so getOrDefault works as intended in
core\addons\functions\Missions\fn_convoy

jwoodruff40 committed:
Cleanup faction hashmaps before using them

jwoodruff40 committed:
Add supply crate retex's for convoys

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Merge branch 'unstable' into patch-50

jwoodruff40 committed:
Replace medical retex's with humanitarian cargo.

jwoodruff40 committed:
Money / Supply convoy container updates

Add money crate retex; use in money convoys (either 1 or 3 on a pallet).

Add pallet and crate to logisitcs.

Rework adding container to vehicle to try large size container first,
then small. Otherwise, objective will still be the truck itself.

jwoodruff40 committed:
Update convoy mission stringtable entries for money / supplies convoy revamp.

jwoodruff40 committed:
Fix rebel faction identity change

wersal454 committed:
Merge pull request #30 from jwoodruff40/pr/wersal454/428

Make money and supply convoy vehicles faction appropriate

wersal454 committed:
Merge branch 'unstable' into patch-50

Silence-112 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -69,57 +73,83 @@ private _taskTitle = "";
-        _taskIcon = "truck";
-        _typeVehObj = selectRandom (FactionGet(reb, "vehiclesCivSupply"));
+        _taskIcon = "takeoff"; ///"truck" icon doesn't exist
+        _vehiclePool = if _civDisabled then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction get "vehiclesCivIndustrial" } select { typeName _x == "STRING"}; // * convert weighted list to normal array
```
```suggestion
        _vehiclePool = if (_civDisabled) then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction get "vehiclesCivIndustrial" } select { typeName _x == "STRING"}; // * convert weighted list to normal array
```

jwoodruff40 replied:
Lgtm; I can't accept the suggestion bc not my PR anymore lol.

I did read on the Arma 3 optimization guide that `select { _x isEqualType "" }` should be significantly faster than `select { typeName _x == "STRING" }` so may be worthwhile to make that change here as well. Or not, idk. probably doesn't realistically make much of a difference.

Silence-112 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -69,57 +73,83 @@ private _taskTitle = "";
-        _taskIcon = "truck";
-        _typeVehObj = selectRandom (FactionGet(reb, "vehiclesCivSupply"));
+        _taskIcon = "box";
+        _vehiclePool = if _civDisabled then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction getOrDefault ["vehiclesCivMedical", _civFaction get "vehiclesCivIndustrial"] } select { typeName _x == "STRING"}; // * convert weighted list to normal array
```
```suggestion
        _vehiclePool = if (_civDisabled) then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction getOrDefault ["vehiclesCivMedical", _civFaction get "vehiclesCivIndustrial"] } select { typeName _x == "STRING"}; // * convert weighted list to normal array
```
Does the else statement (`_civFaction getOrDefault ["vehiclesCivMedical", _civFaction get "vehiclesCivIndustrial"]`) actually work? Shouldn't this be `+` instead of `,`?

jwoodruff40 replied:
It's just an admittedly confusing way to rewrite:

```
_vehiclePool = if (_civDisabled) then {
    _milFaction get "vehiclesMilitiaTrucks"}
else {
    if !(isNil _civFaction get "vehiclesCivMedical") then {
        _civFaction get "vehiclesCivMedical"
    } else {
        _civFaction get "vehiclesCivIndustrial"
    };
};
_vehiclePool = _vehiclePool select { typeNAme _x == "STRING" };
```

My thought process is no need to check if "vehiclesCivIndustrial" is nil, because if the loaded factions don't have *any* of those three keys, then really what we have is a faction template issue, not a code issue.

Also, same comment as above wrt the `select` statement.

Silence-112 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -243,8 +275,67 @@ if (_convoyType isEqualTo "Reinforcements") then
+            private _canLoad = [_vehObj, _supObj] call A3A_Logistics_fnc_canLoad;        
+            if (_canLoad isEqualType -1) then {
+                deleteVehicle _supObj;
+                continue
+            } else {
+                clearMagazineCargoGlobal _supObj;
+                clearWeaponCargoGlobal _supObj;
+                clearItemCargoGlobal _supObj;
+                clearBackpackCargoGlobal _supObj;
+                _supObj setDamage 0.75; // * vanilla supply crates are ridiculously strong. Would make destroying (instead of stealing) the cargo way too hard / resource intensive
+                _supObj lockInventory true; // * don't want pesky inquisitive players to know there's not actually anything in here lol
+                _supObj call A3A_Logistics_fnc_addLoadAction;
+                (_canLoad + [true]) call A3A_Logistics_fnc_load;
+                break
```
Are the `break` and `continue` here really necessary? I can see why the break might be used, but isn't the continue redundant?

jwoodruff40 replied:
So basically what this is doing is checking if the vehicle can load the pallet object (which will have 3 money crate on it), and if not, then whether it can at least load a single money crate.

The `continue` is probably not necessary no. I think it's a holdover from when I was doing other stuff after the if/else statement. The `break` just ensures that if we *can* use the pallet (or any other object, if we add more to the forEach in the future), then we won't waste time checking the rest of the items in the forEach statement and `_supObj` will remain assigned as the pallet (or whatever item we're on). 

There may be a cleaner way to do this. I've dabbled in a lot of different languages for different games and projects (just started with SQF a month or two ago, and learning by reverse engineering lol), but I would call myself somewhere between a novice and amateur haha.

Silence-112 replied:
There's no solution quite like a temporary one

Silence-112 reviewed (commented):
How much has this been tested? Just curious if you've observed any potential problems with edge cases

jwoodruff40 commented:
> How much has this been tested? Just curious if you've observed any potential problems with edge cases

...not enough. But the testing I *did* do, I did at least do with Vanilla, RHS, and random 3CB factions. It needs testing to ensure all types still work as normal. The only ones I actually semi-tested while iterating were the money and supplies ones; however, I know of two issues that need solved before merge:

 - [ ] Wobbling money crates (see my comment in Wersal's repo). I've already tried both `enableSimulation false` and `createSimpleObject` instead of `createVehicle` for the money crates and neither solves the issue.
 - [ ] I don't think I updated the code that deletes / despawns the objective to delete the money crates. I think it will just delete the pallet they sit on, but this will become a non-issue if we have to just load one crate to fix the first issue.

Also the money crate could use a new normal and specular map (whatever Arma calls those)

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Merge branch 'unstable' into patch-50

wersal454 commented:
> > How much has this been tested? Just curious if you've observed any potential problems with edge cases
> 
> ...not enough. But the testing I _did_ do, I did at least do with Vanilla, RHS, and random 3CB factions. It needs testing to ensure all types still work as normal. The only ones I actually semi-tested while iterating were the money and supplies ones; however, I know of two issues that need solved before merge:
> 
> * [ ]  Wobbling money crates (see my comment in Wersal's repo). I've already tried both `enableSimulation false` and `createSimpleObject` instead of `createVehicle` for the money crates and neither solves the issue.
> * [ ]  I don't think I updated the code that deletes / despawns the objective to delete the money crates. I think it will just delete the pallet they sit on, but this will become a non-issue if we have to just load one crate to fix the first issue.
> 
> Also the money crate could use a new normal and specular map (whatever Arma calls those)

Do wobbling crates actually interact with anything or they are "ghostlike" objects with no physical interactions

If they don't affect anything and just wobble from time to time, I think thats kinda cool)

wersal454 commented:
> > > How much has this been tested? Just curious if you've observed any potential problems with edge cases
> > 
> > 
> > ...not enough. But the testing I _did_ do, I did at least do with Vanilla, RHS, and random 3CB factions. It needs testing to ensure all types still work as normal. The only ones I actually semi-tested while iterating were the money and supplies ones; however, I know of two issues that need solved before merge:
> > 
> > * [ ]   Wobbling money crates (see my comment in Wersal's repo). I've already tried both `enableSimulation false` and `createSimpleObject` instead of `createVehicle` for the money crates and neither solves the issue.
> > * [ ]   I don't think I updated the code that deletes / despawns the objective to delete the money crates. I think it will just delete the pallet they sit on, but this will become a non-issue if we have to just load one crate to fix the first issue.
> > 
> > Also the money crate could use a new normal and specular map (whatever Arma calls those)
> 
> Do wobbling crates actually interact with anything or they are "ghostlike" objects with no physical interactions
> 
> If they don't affect anything and just wobble from time to time, I think thats kinda cool)

nuh actually, I think it would be cool if they didn't jitter https://imgur.com/a/wobbling-containers-m3trs1r

with jittering it looks kinda janky, but wobbling effect itself is fine

jwoodruff40 commented:
Yeah I can't figure out how to make them not do that. I think we'll just have to get rid of the pallet and use just one money crate for the time being.

Also, I haven't worked on this pr in a bit, but I think the PR might also be a good opportunity to take a crack at fixing community #3433.

wersal454 committed:
Merge branch 'unstable' into patch-50

wersal454 commented:
> Yeah I can't figure out how to make them not do that. I think we'll just have to get rid of the pallet and use just one money crate for the time being.
> 
> Also, I haven't worked on this pr in a bit, but I think the PR might also be a good opportunity to take a crack at fixing community [official-antistasi-community#3433](https://github.com/official-antistasi-community/A3-Antistasi/issues/3433).

I don't it's possible to fix that issue

Maybe it's possible to add an eventhandler in case marker gets flipped that sends convoy to some other outpost?

Silence-112 commented:
The wobbling crates is due to a funny engine interaction between a chain of attached objects I believe

If I remember correctly, different objects have different "update" rates when things are attached to them. When you chain multiple objects together that have different rates, this issue occurs (network slowdown can also cause it)

You can't necessarily fix it, it would require the crates and pallet to be one object rather than objects attached to each other

wersal454 committed:
Update Stringtable.xml

wersal454 committed:
Update Stringtable.xml

wersal454 commented:
bruh I can't
![image](https://github.com/user-attachments/assets/fd74fb3c-fc31-42f2-8aed-756e670e313a)
I've tried commands in vs code and it says "there is no conflicts, bozo"

wtf

Silence-112 commented:
@wersal454 If you have [git](https://git-scm.com/downloads) installed (not github), right click in the root A3 repo and click "Open Git Bash here"

Easiest way (for me, at least) is to run:
`git remote add SilenceIsFatto https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate`
`git fetch SilenceIsFatto`
`git checkout patch-50`
`git merge SilenceIsFatto/unstable`

It should then show which files have a merge conflict and you can resolve them in vscode and commit as usual

wersal454 commented:
"Merge conflict in A3A/addons/scrt/Stringtable.xml"

Exactly what github tells me

wersal454 committed:
Merge branch 'unstable' into patch-50

wersal454 committed:
Update Stringtable.xml

<Original>A truck with humanitarian supplies destination %3 is about to depart at %2 from %1. Steal those supplies, bring them to %3 and let people in there know it is %4 who's giving those supplies.</Original>

wersal454 committed:
Merge branch 'patch-50' of https://github.com/wersal454/A3-Antistasi into patch-50

wersal454 committed:
Merge branch 'unstable' into patch-50

wersal454 committed:
Update Stringtable.xml

wersal454 committed:
Merge branch 'patch-50' of https://github.com/wersal454/A3-Antistasi into patch-50

wersal454 commented:
Jesus

finally

wersal454 committed:
Merge branch 'unstable' into patch-50

wersal454 commented:
> Yeah I can't figure out how to make them not do that. I think we'll just have to get rid of the pallet and use just one money crate for the time being.
> 
> Also, I haven't worked on this pr in a bit, but I think the PR might also be a good opportunity to take a crack at fixing community [official-antistasi-community#3433](https://github.com/official-antistasi-community/A3-Antistasi/issues/3433).

If there isn't a bigger crate that could be retextured, current one will do

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -246,8 +278,67 @@ if (_convoyType isEqualTo "Reinforcements") then
+{
+    if (_objectiveIsCargo) then {
+        // * put a supply container in the truck so it can be identified more easily as the objective vehicle
+        {
+            _supObj = _x createVehicle (position _vehObj);
+            if (_forEachIndex == 0) then {
+                // * attach money crates to pallet
+                private _crateOffset = [0.52, 0.02, -0.48];
+                for "_i" from 0 to 2 do {
+                    private _crate = "A3AU_moneyCrate_small_01" createVehicle (position _supObj);
+                    _crate lockInventory true;
+                    _crate attachTo [_supObj, [0, _crateOffset select _i, 0.29]];
+                };
+            };
+            
+            // * try to load a large container, then fall back to small box if we can't load large container
+            private _canLoad = [_vehObj, _supObj] call A3A_Logistics_fnc_canLoad;        
+            if (_canLoad isEqualType -1) then {
+                deleteVehicle _supObj;
+                continue
+            } else {
+                clearMagazineCargoGlobal _supObj;
+                clearWeaponCargoGlobal _supObj;
+                clearItemCargoGlobal _supObj;
+                clearBackpackCargoGlobal _supObj;
+                _supObj setDamage 0.75; // * vanilla supply crates are ridiculously strong. Would make destroying (instead of stealing) the cargo way too hard / resource intensive
+                _supObj lockInventory true; // * don't want pesky inquisitive players to know there's not actually anything in here lol
+                _supObj call A3A_Logistics_fnc_addLoadAction;
+                (_canLoad + [true]) call A3A_Logistics_fnc_load;
+                break
+            };
+        } forEach ["Land_Pallet_F", "A3AU_moneyCrate_small_01"];
```
Remove the extra money crates and pallet. Just use one.

```suggestion
            // * put a supply container in the truck so it can be identified more easily as the objective vehicle
        _supObj = "A3AU_moneyCrate_small_01" createVehicle (position _vehObj);
        
        private _canLoad = [_vehObj, _supObj] call A3A_Logistics_fnc_canLoad;        
        if (_canLoad isEqualType []) then {
            clearMagazineCargoGlobal _supObj;
            clearWeaponCargoGlobal _supObj;
            clearItemCargoGlobal _supObj;
            clearBackpackCargoGlobal _supObj;
            _supObj setDamage 0.75;
            _supObj lockInventory true; // * don't want pesky inquisitive players to know there's not actually anything in here lol
            _supObj call A3A_Logistics_fnc_addLoadAction;
            (_canLoad + [true]) call A3A_Logistics_fnc_load;
        };
```

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -72,57 +76,83 @@ private _taskTitle = "";
         _taskIcon = "run";
-        _typeVehObj = selectRandom (_faction get "vehiclesTrucks");
+        _typeVehObj = selectRandom ( switch true do {
+            case (tierWar < 5): { (_milFaction get "vehiclesMilitiaTrucks") };
+            case (tierWar < 7): { (_milFaction get "vehiclesMilitiaTrucks") + (_milFaction get "vehiclesTrucks") };
+            default { (_milFaction get "vehiclesTrucks") };
```
Can we use the tier war levels configured in Antistasi CBA settings instead of statically setting them to 5/7?
![cba settings](https://github.com/user-attachments/assets/0b75b27f-2338-45f7-90bf-67637b3f16f6)

Also, why are these CBA settings and not params? @SilenceIsFatto specifically the war level related settings

Silence-112 replied:
Because the params don't have a native slider control

jwoodruff40 replied:
Hmm, I can probably add sliders to certain params instead of LBs. At least to see if it feels better. It could make it feel more disjointed though, one way to find out.

wersal454 replied:
I don't think that that should used for this, those sliders are for unit types purely, vehicle stuff is unaffected iIrc

Also, other missions also don't use them or affected by them (again iIrc)

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -72,57 +76,83 @@ private _taskTitle = "";
         _taskIcon = "meet";
-        _typeVehObj = selectRandom (_faction get "vehiclesTrucks");
+        _typeVehObj = selectRandom ( switch true do {
+            case (tierWar < 5): { (_milFaction get "vehiclesMilitiaTrucks") };
+            case (tierWar < 7): { (_milFaction get "vehiclesMilitiaTrucks") + (_milFaction get "vehiclesTrucks") };
+            default { (_milFaction get "vehiclesTrucks") };
```
same question about the cba settings

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -72,57 +76,83 @@ private _taskTitle = "";
-        _taskIcon = "truck";
-        _typeVehObj = selectRandom (FactionGet(reb, "vehiclesCivSupply"));
+        _taskIcon = "takeoff"; ///"truck" icon doesn't exist
+        _vehiclePool = if (_civDisabled) then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction get "vehiclesCivIndustrial" } select { typeName _x == "STRING"}; // * convert weighted list to normal array
```
```suggestion
        _vehiclePool = if (_civDisabled) then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction get "vehiclesCivIndustrial" } select { _x isEqualType "" }; // * convert weighted list to normal array
```

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -72,57 +76,83 @@ private _taskTitle = "";
-        _taskIcon = "truck";
-        _typeVehObj = selectRandom (FactionGet(reb, "vehiclesCivSupply"));
+        _taskIcon = "box";
+        _vehiclePool = if (_civDisabled) then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction getOrDefault ["vehiclesCivMedical", _civFaction get "vehiclesCivIndustrial"] } select { typeName _x == "STRING"}; // * convert weighted list to normal array
```
```suggestion
        _vehiclePool = if (_civDisabled) then { _milFaction get "vehiclesMilitiaTrucks" } else { _civFaction getOrDefault ["vehiclesCivMedical", _civFaction get "vehiclesCivIndustrial"] } select { _x isEqualType "" }; // * convert weighted list to normal array
```

jwoodruff40 commented on A3A/addons/core/functions/Missions/fn_convoy.sqf:
```diff
@@ -265,7 +356,12 @@ for "_i" from 1 to _countX do
-private _typeVehX = selectRandom (if (_sideX == Occupants && random 4 < tierWar) then {_faction get "vehiclesPolice"} else {_faction get "vehiclesLightArmed"});
+private _typeVehX = selectRandom ( switch true do {
+    case (_sideX == Occupants && random 4 < tierWar): { (_milFaction get "vehiclesPolice") };
+    case (tierWar < 5): { (_milFaction get "vehiclesMilitiaLightArmed") };
+    case (tierWar < 7): { (_milFaction get "vehiclesLightArmed") + (_milFaction get "vehiclesMilitiaLightArmed") };
+    default { (_milFaction get "vehiclesLightArmed") };
```
same question about the cba settings

jwoodruff40 commented on A3A/addons/core/CfgVehicles.hpp:
```diff
@@ -148,42 +148,67 @@ class CfgVehicles
+        class TransportBackpacks{};
+    };
+
+    class B_supplyCrate_F;
+    class A3AU_supplyCrate_Base : B_supplyCrate_F {
+        author = AUTHOR;
+        class TransportBackpacks {};
+        class TransportItems {};
+        class TransportMagazines {};
+        class TransportWeapons {};
+    };
+    class A3AU_supplyCrate_Medical_01 : A3AU_supplyCrate_Base {
+        author = AUTHOR;
+        hiddenSelections[] = { "camo" };
+		hiddenSelectionsTextures[] = { QPATHTOFOLDER(Pictures\items\supplydrop_medical_01_co.paa) };
+    };
+    class A3AU_supplyCrate_Medical_02 : A3AU_supplyCrate_Base {
+        author = AUTHOR;
+        hiddenSelections[] = { "camo" };
+		hiddenSelectionsTextures[] = { QPATHTOFOLDER(Pictures\items\supplydrop_medical_02_co.paa) };
+    };
+    
```
These aren't used anymore 
```suggestion
```

jwoodruff40 commented on A3A/addons/core/Pictures/Items/supplydrop_medical_01_co.paa:
delete this

jwoodruff40 commented on A3A/addons/core/Pictures/Items/supplydrop_medical_02_co.paa:
delete this too

jwoodruff40 reviewed (changes requested):
Left some comments to deal with the money crate thing and some other little stuff.

Wth is all this doing in here though?
![wth](https://github.com/user-attachments/assets/b8beebcc-490a-4b3b-85af-98840d3d2053)

Might need to do an interactive rebase on unstable to get rid of all that. Probably an artifact of your repo being forked from community instead of ultimate and a bad merge.

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_convoy.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/CfgVehicles.hpp

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>

wersal454 committed:
delete paa

wersal454 committed:
Merge branch 'unstable' into patch-50

Silence-112 commented:
Continued in #524
